### PR TITLE
Documentation tweaks

### PR DESCRIPTION
**Pull request by jwass**
A few wording changes - mostly at the top of a few files.

### CODE REVIEW

File: src/collections/mod.rs [hunk 1]
```diff
@@ -1,6 +1,4 @@
-//! This module contains the parallel iterator types for standard
-//! collections. You will rarely need to interact with it directly
-//! unless you have need to name one of the iterator types.
+//! Parallel iterator types for [standard collections](std::collections).
```
Comment (KodrAus) [inline src/collections/mod.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [standard collections](std::collections).
//! 
//! [std::collections]: https://doc.rust-lang.org/std/collections/
```

File: src/option.rs [hunk 2]
```diff
@@ -1,6 +1,4 @@
-//! This module contains the parallel iterator types for options
-//! (`Option<T>`). You will rarely need to interact with it directly
-//! unless you have need to name one of the iterator types.
+//! Parallel iterator types for [options](std::option).
```
Comment (KodrAus) [inline src/option.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [options](std::option).
//! 
//! [std::option]: https://doc.rust-lang.org/std/option/
```

File: src/collections/mod.rs (hunk 1 above)
Comment (cuviper) [inline src/collections/mod.rs]: I think this style of auto-link only works with nightly.  Can you manually link it?

File: src/range.rs [hunk 3]
```diff
@@ -1,7 +1,5 @@
-//! This module contains the parallel iterator types for ranges
-//! (`Range<T>`); this is the type for values created by a `a..b`
-//! expression. You will rarely need to interact with it directly
-//! unless you have need to name one of the iterator types.
+//! Parallel iterator types for [ranges](std::ops::Range), the type
```
Comment (KodrAus) [inline src/range.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [ranges](std::ops::Range), the type
//! for values created by `a..b` expressions.
//! 
//! [std::ops::Range]: https://doc.rust-lang.org/std/ops/struct.Range.html
```

File: src/result.rs [hunk 4]
```diff
@@ -1,6 +1,4 @@
-//! This module contains the parallel iterator types for results
-//! (`Result<T, E>`). You will rarely need to interact with it directly
-//! unless you have need to name one of the iterator types.
+//! Parallel iterator types for [results](std::Result).
```
Comment (KodrAus) [inline src/result.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [results](std::result).
//! 
//! [std::result]: https://doc.rust-lang.org/std/result/
```

File: src/slice/mod.rs [hunk 5]
```diff
@@ -1,6 +1,4 @@
-//! This module contains the parallel iterator types for slices
-//! (`[T]`). You will rarely need to interact with it directly unless
-//! you have need to name one of those types.
+//! Parallel iterator types for [slices](std::slice).
```
Comment (KodrAus) [inline src/slice/mod.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [slices](std::slice).
//! 
//! [std::slice]: https://doc.rust-lang.org/std/slice/
```

File: src/str.rs [hunk 6]
```diff
@@ -1,8 +1,4 @@
-//! This module contains extension methods for `String` that expose
-//! parallel iterators, such as `par_split_whitespace()`. You will
-//! rarely need to interact with it directly, since if you add `use
-//! rayon::prelude::*` to your file, that will include the helper
-//! traits defined in this module.
+//! Parallel iterator types for [strings](std::str).
```
Comment (KodrAus) [inline src/str.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [strings](std::str).
//!
//! Note: [`ParallelString::par_split()`] and [`par_split_terminator()`]
//! reference a `Pattern` trait which is not visible outside this crate.
//! This trait is intentionally kept private, for use only by Rayon itself.
//! It is implemented for `char` and any `F: Fn(char) -> bool + Sync + Send`.
//!
//! [`ParallelString::par_split()`]: trait.ParallelString.html#method.par_split
//! [`par_split_terminator()`]: trait.ParallelString.html#method.par_split_terminator
//! [std::str]: https://doc.rust-lang.org/std/str/
```

File: src/vec.rs [hunk 7]
```diff
@@ -1,6 +1,4 @@
-//! This module contains the parallel iterator types for vectors
-//! (`Vec<T>`). You will rarely need to interact with it directly
-//! unless you have need to name one of those types.
+//! Parallel iterator types for [vectors](std::vec).
```
Comment (KodrAus) [inline src/vec.rs]: We'll need to add a markdown link for each of these new links to std we've introduced so they light up in the rendered docs:

```rust
//! Parallel iterator types for [vectors](std::vec).
//! 
//! [std::vec]: https://doc.rust-lang.org/std/vec/
```

**Comment by jwass** [comment]
@KodrAus Fixed up the links from your comments.

**Comment by cuviper** [comment]
bors delegate=KodrAus

File: src/range.rs [hunk 8]
```diff
@@ -12,6 +10,8 @@
 //! // compare result with sequential calculation
 //! assert_eq!((0..100).sum::<u64>(), r);
 //! ```
+//!
+//! [std::range]: https://doc.rust-lang.org/stable/std/vec/
```
Comment (KodrAus) [inline src/range.rs]: Looks we've got the wrong link here.

**Comment by jwass** [comment]
@KodrAus This is ready to go. Plus I added back in the lines about not needing to indirectly directly with modules included in the top level.

**Comment by KodrAus** [comment]
Thanks @jwass! This looks good to me.

bors r+

**Pull request merged**